## Очистка ресурсов (#3)

Цель этого упражнения — проверить, что, независимо от того, как ваш код выходит из блока `use()`, функция `close()` всегда вызывается. Используйте класс `Cleanup`, который вы создали в упражнении 2 и который включен в начальный код, вместе с `enum class Option { Normal, Return, Throw }`.

Определите функцию `verifyClose(opt: Option)`. `verifyClose()` создает объект `Cleanup` и вызывает `use()` на нем. Лямбда для `use()` содержит выражение `when`, которое реализует эти правила. Когда `opt`:

- `Normal`: вызывается `f()`, затем `g()` (внутри выражения `when`), затем `h()`, перед выходом из блока `use()`.

- `Return`: вызывается `f()`, а затем `return`. Это завершает выполнение из середины блока `use()`.

- `Throw`: вызывается `f()`, а затем исключение `Exception` выбрасывается из середины блока `use()`.

`main()` тестирует ваш код.

<sub> Это задание не содержит автоматических тестов, поэтому оно всегда помечается как "Правильно", когда вы запускаете "Проверку". Пожалуйста, сравните ваше решение с предоставленным! </sub>